rocAL/rocAL/source/pipeline/master_graph.cpp

```diff
@@ -1070,6 +1090,16 @@ void MasterGraph::output_routine()
                 else
                     full_batch_meta_data = _augmented_meta_data->clone();
             }
+            // get roi width and height of output image
```

Comment:
I don't understand the logic here. Why are we using another array for resize_width and resize_height? _output_image_info assumed to be constant for all the images and reinitializing every time here won't change anything for resize_width and height. In this case it is better to use output_image_info directly in the resize_width and resize_height api calls

Reply:
The reasoning behind this implementation was that the _resize_width and _resize_height acts as a queue which inserts the ROI info of each batch in a new vector at the front. The output_resize_width function in the same file pops the last ROI vector and returns it to the vector. Due to prefetch_queue_depth, we couldn't directly fetch the ROI info from ouptut_image_info and return it to the user. I tried that as part of the experiments to fix the HIP backend issue and it gave wrong values.
The fix for the HIP backend mask training removes this vector updation and the respective ROI API functions. Instead I'm adding the ROI as an internal ImgSize struct variable of MetaDataBatch and updating it in meta_node for RMN. I'll be pushing the new ROI implementation soon after the internal review